src/Workspaces/Core/Portable/Classification/IClassificationService.cs

```diff
@@ -50,13 +51,13 @@ internal interface IClassificationService : ILanguageService
         /// This will not include classifications for embedded language constructs in string literals.  For that use
         /// <see cref="AddEmbeddedLanguageClassificationsAsync"/>.
         /// </remarks>
-        Task AddSemanticClassificationsAsync(Document document, TextSpan textSpan, ClassificationOptions options, ArrayBuilder<ClassifiedSpan> result, CancellationToken cancellationToken);
+        Task AddSemanticClassificationsAsync(Document document, TextSpan textSpan, ClassificationOptions options, ImmutableSegmentedList<ClassifiedSpan>.Builder result, CancellationToken cancellationToken);
```

Comment:
📝 The other methods in this interface weren't showing the same behavior. If we decide to proceed with this change, it might make sense to update all of the methods for consistency.